stellargraph/random.py

```diff
@@ -75,3 +76,31 @@ def set_seed(seed):
         _rs = _global_state()
     else:
         _rs = _seeded_state(seed)
+
+
+class SeededPerBatch:
```

Comment:
Is it worth having some tests for this? E.g. checking that the seeds are always in the same order no matter the order of `batch_num` queries or something?

Reply:
Sure, I'll add one 👍